gtda/images/filtrations.py

```diff
@@ -1036,3 +1028,227 @@ def plot(Xt, sample=0, colorscale='greys', origin='upper',
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    mask_ : ndarray of shape (radius, radius, [, radius])
+        The mask applied around each pixel to calculate the weighted number of
+        its activated neighbors. Set in :meth:`fit`.
+
+    See also
+    --------
+    gtda.homology.CubicalPersistence, Binarizer
+
+    References
+    ----------
+    [1] A. Garin and G. Tauzin, "A topological reading lesson: Classification
+        of MNIST  using  TDA"; 19th International IEEE Conference on Machine
+        Learning and Applications (ICMLA 2020), 2019; arXiv: `1910.08345 \
+        <https://arxiv.org/abs/1910.08345>`_.
+
+    """
+
+    _hyperparameters = {
+        'radius': {'type': Real, 'in': Interval(0, np.inf, closed='right')},
+        'metric': {'type': (str, FunctionType)},
+        'metric_params': {'type': dict},
+    }
+
+    def __init__(self, radius=3, metric='euclidean', metric_params={},
+                 n_jobs=None):
+        self.radius = radius
+        self.metric = metric
+        self.metric_params = metric_params
+        self.n_jobs = n_jobs
+
+    def _calculate_density(self, X):
+        Xd = np.zeros(X.shape)
+
+        for i, j, k in self._iterator:
+            Xd += np.roll(np.roll(
+                np.roll(X, k, axis=3), j, axis=2), i, axis=1) \
+                * self.mask_[self._range + i, self._range + j,
+                             self._range + k]
+        return Xd
+
+    def fit(self, X, y=None):
+        """Calculate :attr:`mask_` from a collection of binary images. Then,
+        return the estimator.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_pixels_x, n_pixels_y [, n_pixels_z])
+            Input data. Each entry along axis 0 is interpreted as a 2D or 3D
+            binary image.
+
+        y : None
+            There is no need of a target in a transformer, yet the pipeline API
+            requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        X = check_array(X, allow_nd=True)
+        self._n_dimensions = X.ndim - 1
+        if (self._n_dimensions < 2) or (self._n_dimensions > 3):
+            warn(f"Input of `fit` contains arrays of dimension "
+                 f"{self._n_dimensions}.")
+        validate_params(
+            self.get_params(), self._hyperparameters, exclude=['n_jobs'])
+
+        self._range = int(np.ceil(self.radius))
+
+        iterator_range_list = [range(-self._range, self._range + 1)
+                               for _ in range(self._n_dimensions)] \
+            + [[0] for _ in range(3 - self._n_dimensions)]
```

Comment:
Related to the question above: if the `3` is hardcoded, then `self._n_dimensions` must be `2` and there seems to be no point in defining it.

Reply:
```suggestion
            + [[0] * (3 - self._n_dimensions)]
```
or even just `[[0]]` depending on what you think about my previous point.